File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int rCount = 0;
        int lCount = 0;
        int answer = 0;
        char[] chars = s.toCharArray();
        for (char c : chars) {
            if (c == 'R') rCount++;
            else if (c == 'L') lCount++;
            if (rCount == lCount) {
                answer++;
                rCount = 0;
                lCount = 0;
            }
        }
        return answer;
    }
}

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.kt
class Solution {
fun balancedStringSplit(s: String): Int {
    var count = 0
    var balance = 0
    for (c in s) {
        if (c == 'L') {
            balance++
        } else {
            balance--
        }
        if (balance == 0) {
            count++
        }
    }
    return count
}
}